### DISCUSSION TITLE
Crosshair for data or data region selection

### DISCUSSION THREAD

**Discussion by permiakover** [HanSolo/charts#55] [Ideas]
Hello! Is it supposed to be an instrument to make a selections over charts with a mouse?

**Comment by HanSolo** [comment]
Could you be a bit more specific please? Which chart? What should be selected?

**Reply by permiakover** [reply]
for example, I have a linechart, i would like to have some vertical or/and horizontal line that follows the mouse cursor, highlighting the value (x or y, or both) under the mouse cursor. the value is dedicated to value from xyseries

**Comment by HanSolo** [comment]
That will be possible, implemented that in some other controls already…will take a look at it but might take some time cause I’m on vacation 😁

**Reply by permiakover** [reply]
cool! there is also a good case to select a region with a couple of such markers, that stuff works fine in labview virtual devices.

**Comment by HanSolo** [comment]
I've took a quick look at the stuff today but it seems to be more complex than I thought because I mainly use the canvas in the charts lib. To be able to get the functionality you've asked for I need to add some shadow paths/polygons that keep each data series. It also won't work for each chart type, best candidate would be area charts and maybe linecharts.

**Reply by permiakover** [reply]
hi, any updates for plans about this feature?

**Comment by HanSolo** [comment]
I've commited a first version to the jdk16 branch (commit hash: 37a4c277e515503c9a824499b02e8081fb35dde5).
I've implemented an additional canvas in XYPane which will only be visible if enabled (xyPane.setCrossHairVisible(true)). If it is visible it will fire CursorEvents which contain the x- and y-value of the current mouse position in the chart. You will find an example in the LineChartTest (rows 165 - 182) where I show a tooltip with the current values when the mouse was released.
Please let me know if that is something you can work with...

**Reply by permiakover** [reply]
thank, you! i will test in next few days!
